### PR TITLE
chore: aligns function descriptions with godoc standards in the types package

### PR DESCRIPTION
**Pull request by staheri14**
This PR resolves inconsistencies in the `types` package, particularly in functions where their descriptions either do not follow the Godoc convention (where a comment describing a function should start with the function's name) or exhibit discrepancies with the actual function name.

#### PR checklist

- [ ] Tests written/updated
- [ ] Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)
- [ ] Updated relevant documentation (`docs/` or `spec/`) and code comments

### CODE REVIEW

File: types/tx_test.go [hunk 1]
```diff
@@ -123,7 +123,7 @@ func testTxProofUnchangable(t *testing.T) {
 	}
 }
 
-// This makes sure that the proof doesn't deserialize into something valid.
+// assertBadProof makes sure that the proof doesn't deserialize into something valid.
```
Comment (thanethomson) [inline types/tx_test.go:126]: We don't follow this convention for private methods.

**Review by thanethomson — COMMENTED** [review]
Thanks for this @staheri14!

**Review by melekes — APPROVED** [review]
👍

File: types/tx_test.go (hunk 1 above)
Comment (adizere) [reply types/tx_test.go:126]: Does this change need to be reverted, or can we accept the PR despite this change?

Seems good to me to have the consistency in comments, regardless of this being private method.

**Review by adizere — APPROVED** [review]

**Review by lasarojc — APPROVED** [review]

**Review by thanethomson — APPROVED** [review]

File: types/tx_test.go (hunk 1 above)
Comment (thanethomson) [reply types/tx_test.go:126]: It's fine, it's just inconsistent with the rest of the codebase, which isn't so great.

Comment (staheri14) [reply types/tx_test.go:126]: Thank you for providing clarification, @thanethomson! Regarding consistency within the codebase, I've noticed that there are instances where private methods adhere to this convention [here](https://github.com/cometbft/cometbft/blob/38a4caeac0551c188af8a3e209e48380cd514e2d/statesync/chunks.go#L43). However, if it has been decided that this convention should not be followed strictly in the cometbft codebase, I will take note of this for future PRs.

**Pull request merged**